pie/storage/database.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from typing import Dict, Optional, Union
+
+from sqlalchemy import BigInteger, Column, String
+
+from pie.database import database, session
+
+
+class StorageData(database.base):
+    __tablename__ = "pie_storage_data"
+
+    module = Column(String, primary_key=True)
+    guild_id = Column(BigInteger, primary_key=True)
+    key = Column(String, primary_key=True)
+    value = Column(String)
+    type = Column(String)
+
+    @staticmethod
+    def set(
+        module: str,
+        guild_id: int,
+        key: str,
+        value,
+        allow_overwrite: bool = True,
+    ) -> Optional[StorageData]:
+        data = (
+            session.query(StorageData)
+            .filter_by(module=module)
+            .filter_by(guild_id=guild_id)
+            .filter_by(key=key)
+            .one_or_none()
+        )
+
+        if data and not allow_overwrite:
+            return None
+
+        if not data:
+            data = StorageData(module=module, key=key, guild_id=guild_id)
+
+        data.value = value
+        data.type = type(value).__name__
+        session.merge(data)
+        session.commit()
+
+        return data
+
+    @staticmethod
+    def get(module: str, guild_id: int, key: str) -> Optional[StorageData]:
+        data = (
+            session.query(StorageData)
+            .filter_by(module=module)
+            .filter_by(key=key)
+            .filter_by(guild_id=guild_id)
+            .one_or_none()
+        )
+        return data
+
+    def delete(self):
+        session.delete(self)
+        session.commit()
```

Comment:
Through the codebase, we've been using
```py
@classmethod
def remove(cls, module: str, guild_id: int, key: str) -> bool:
    count = session.get(cls).filter_by(module=module, guild_id=guild_id, key=key).delete()
    return count == 1
```
This means that you don't have to perform `.get()` to check if the value exists in the first place, you can just check the return code.

This would make your `.unset()` method shorter, too.

Reply:
Sounds reasonable, fixed.